test/e2e/upgrade_test.go

```diff
@@ -241,3 +253,32 @@ func (tc *testContext) testWorkloadsAvailable(t *testing.T) {
 		})
 	assert.NoError(t, err)
 }
+
+// waitForCSINodes waits for a CSINode to exist for each node with a driver loaded, indicating CSI functionality has
+// been enabled for the given node.
+func (tc *testContext) waitForCSINodesWithDrivers(nodes []v1.Node) error {
+	return wait.PollImmediate(retry.Interval, 10*time.Minute, func() (bool, error) {
```

Comment:
I feel [retry.Timeout ](https://github.com/openshift/windows-machine-config-operator/blob/833c1b6b6c24c02de9ac68e7f903d092b8b889aa/pkg/retry/retry.go#L13) may be a good fit here.

Reply:
agreed, should have done that in the original PR